### PR TITLE
Add Cilium kube-proxy replacement support

### PR DESCRIPTION
**Pull request by chrisdoherty4**
Cilium can take ownership of all kube-proxy responsibilities. In Cilium v1.13 there are partial and strict modes, however the API changes in v1.14 to a binary setting so this support is for v1.13 strict mode only.

### CODE REVIEW

File: pkg/providers/cloudstack/config/template-cp.yaml [hunk 1]
```diff
@@ -306,6 +306,12 @@ spec:
           timeAdded: {{ .TimeAdded }}
 {{- end }}
 {{- end }}
+{{- end }}
+{{- with .kubeadmSkipPhases }}
+      skipPhases:
+      {{- range . }}
+      - {{ . }}
+      {{- end }}
```
Comment (cxbrowne1207) [inline pkg/providers/cloudstack/config/template-cp.yaml:314]: Curious. Why was this needed?

Comment (chrisdoherty4) [reply pkg/providers/cloudstack/config/template-cp.yaml:314]: Why is `skipPhases` needed or why is `end`?

Comment (cxbrowne1207) [reply pkg/providers/cloudstack/config/template-cp.yaml:314]: `skipPhases`
